astropy/utils/data_info.py

```diff
@@ -514,6 +519,60 @@ def slice_indices(self, col_slice, item, col_len):
 
         return col_slice
 
+    @staticmethod
+    def merge_cols_attributes(cols, metadata_conflicts, name, attrs):
+        """
+        Utility method to merge and validate the attributes ``attrs`` for the
+        input table columns ``cols``.
+
+        Note that ``dtype`` and ``shape`` attributes are handled specially.
+        These should not be passed in ``attrs`` but will always be in the
+        returned dict of merged attributes.
+
+        Parameters
+        ----------
+        cols : list
+            List of input Table column objects
+        metadata_conflicts : str ('warn'|'error'|'silent')
+            How to handle metadata conflicts.
+        name : str
+            Output column name
+        attrs : list
+            List of attribute names to be merged
+
+        Returns
+        -------
+        attrs : dict of merged attributes
+
+        """
+        from ..table.np_utils import TableMergeError
+
+        def warn_str_func(key, left, right):
+            out = ("In merged column '{}' the '{}' attribute does not match "
+                   "({} != {}).  Using {} for merged output"
+                   .format(name, key, left, right, right))
+            return out
+
+        def getattrs(col):
+            return {attr: getattr(col.info, attr) for attr in attrs
```

Comment:
I think the clarity provided by the functional expression is worth the double getattr.